sormas-backend/src/main/java/de/symeda/sormas/backend/common/ChangeDateFilterBuilder.java

```diff
@@ -24,21 +25,32 @@
 import javax.persistence.criteria.JoinType;
 import javax.persistence.criteria.Predicate;
 
+import de.symeda.sormas.api.EntityDto;
+
 public class ChangeDateFilterBuilder {
 
 	private final CriteriaBuilder cb;
 	private final Stream.Builder<Predicate> filters;
-	private Date date;
+	private Timestamp date;
 	private Expression<? extends Date> dateExpression;
 
+	private From<?, ?> root;
+	private String lastSynchronizedUuidSameTimestamp;
+
 	private ChangeDateFilterBuilder(CriteriaBuilder cb) {
 		this.cb = cb;
 		this.filters = Stream.builder();
 	}
 
 	public ChangeDateFilterBuilder(CriteriaBuilder cb, Date date) {
 		this(cb);
-		this.date = date;
+		this.date = new Timestamp(date.getTime());
```

Comment:
The date is passed in as timestamp (e.g. ContactService), but here only interpreted as date and then again cast to timestamp. Quite confusing to me. I think the date field should keep the Date type.

Reply:
The conversation to Timestamp is necessary and should be computed only once. Rename the date field?

Reply:
Refactored to use `timestampUpper` and `timestampLower`.